app/models/donor.rb

```diff
@@ -5,8 +5,8 @@ class Donor < ApplicationRecord
   validates :email, uniqueness: true, allow_nil: true
   before_validation { self.email = nil if email.blank? }
 
-  has_many :donations
-  has_many :donor_addresses
+  has_many :donations, dependent: :destroy
```

Comment:
This scares me a bit.... we could accidentally delete a bunch of donations here if we delete 1 donor not thinking about the amount of donations attached to them.... I'd rather not do that on this relation.